### PR TITLE
Story/geco 122

### PR DESCRIPTION
**Pull request by abhis1989kumar**

### CODE REVIEW

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/core/service/impl/OCRManager.java [hunk 1]
```diff
@@ -42,7 +42,6 @@
 @Service
 public class OCRManager implements IOCRManager {
 
-    private final Logger logger = LoggerFactory.getLogger(getClass());
```
Comment (jdamerow) [inline cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/core/service/impl/OCRManager.java:45]: if you remove the logger, please also remove the imports

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java [hunk 2]
```diff
@@ -36,12 +39,49 @@
...
         validateBinder.addValidators(new SystemConfigValidator());
         ocrTypeMap.put(Properties.OCR_PLAINTEXT, propertyManager.getProperty(Properties.OCR_PLAINTEXT));
         ocrTypeMap.put(Properties.OCR_HOCR, propertyManager.getProperty(Properties.OCR_HOCR));
+        
+        String[] langs = getTessLangs();
+        for(int i=1;i<langs.length;i++) {
+        	langTypeMap.put(langs[i],langs[i]);
+        } 
+        if(langTypeMap.containsKey(Properties.ENGLISH)) {
+        	defaultLang = langTypeMap.get(Properties.ENGLISH);
```
Comment (jdamerow) [inline cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:57]: and what if not?

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java [hunk 3]
```diff
@@ -36,12 +39,49 @@
     private ISystemMessageHandler messageHandler;
 
     Map<String, String> ocrTypeMap = new HashMap<>();
+    Map<String, String> langTypeMap = new HashMap<>();
+    String defaultLang = new String();
```
Comment (jdamerow) [inline cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:43]: those should be private (ocrTypeMap as well actually)

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java [hunk 4]
```diff
@@ -36,12 +39,49 @@
...
+	        proc.waitFor();   
+		} catch (IOException e) {
+            messageHandler.handleMessage("Error while getting the Tesserract laguages.", e, MessageType.ERROR);
+		} catch (InterruptedException e) {
+            messageHandler.handleMessage("Error while getting the Tesserract laguages.", e, MessageType.ERROR);
+		}
+        lang_list = output.split(":");
+        languages = lang_list[1].split(" ");
+        
+        return languages;
```
Comment (jdamerow) [inline cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:84]: this method does not belong into the controller. there are classes handling tesseract connections. controllers are only meant for transferring data between view and business logic.

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java [hunk 5]
```diff
@@ -36,12 +39,49 @@
...
+        for(int i=1;i<langs.length;i++) {
+        	langTypeMap.put(langs[i],langs[i]);
+        } 
+        if(langTypeMap.containsKey(Properties.ENGLISH)) {
+        	defaultLang = langTypeMap.get(Properties.ENGLISH);
+        }
+    }
+    
+    public String[] getTessLangs() {
+        String command = "/usr/local/bin/tesseract --list-langs";
```
Comment (jdamerow) [inline cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:62]: depending on which system we're running on, this path could be anything.

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java (hunk 4 above)
Comment (jdamerow) [reply cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:84]: also, formatting is all messed up.

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java [hunk 6]
```diff
@@ -36,12 +39,49 @@
...
+			reader =  new BufferedReader(new InputStreamReader(proc.getInputStream()));
+			String line = "";
+	        while((line = reader.readLine()) != null) {
+	        	output = output + line + " ";
+	        }
+	        proc.waitFor();   
+		} catch (IOException e) {
+            messageHandler.handleMessage("Error while getting the Tesserract laguages.", e, MessageType.ERROR);
+		} catch (InterruptedException e) {
+            messageHandler.handleMessage("Error while getting the Tesserract laguages.", e, MessageType.ERROR);
```
Comment (jdamerow) [inline cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:79]: laguages

Comment (jdamerow) [reply cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:79]: also should be "getting Tesseract..." not "getting the Tesseract..."

**Review by jdamerow — CHANGES_REQUESTED** [review]
fix formatting.

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/core/service/impl/OCRManager.java (hunk 1 above)
Comment (abhis1989kumar) [reply cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/core/service/impl/OCRManager.java:45]: Removed the unused imports.

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java (hunk 2 above)
Comment (abhis1989kumar) [reply cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java]: Currently if the list of languages doesn't have English, it populates the first language in the list such as arabic. Is there anything else I need to change in this case? Kindly let me know...will add the logic accordingly.

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java (hunk 3 above)
Comment (abhis1989kumar) [reply cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:43]: Made  ocrTypeMap and langTypeMap private Maps

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java (hunk 6 above)
Comment (abhis1989kumar) [reply cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:79]: Fixed

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java (hunk 2 above)
Comment (abhis1989kumar) [reply cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:57]: The logic picks the first language from the list if the lang English is not present.Also empty space is displayed if the language-list is empty.

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java (hunk 4 above)
Comment (abhis1989kumar) [reply cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:84]: Moved getTessLangs() method to the existing TesseractOCRParser.java class

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java (hunk 5 above)
Comment (abhis1989kumar) [reply cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java:62]: Changed the logic; now taking the path value from Properties.

**Pull request closed without merging**